### ISSUE TITLE
`ignore next` above object-literal method doesn't suppress branches in the method body

### ISSUE THREAD

**Issue by simleb** [fallow-rs/oxc-coverage-instrument#20]
A `/* v8 ignore next */` pragma placed directly above a method in an object literal removes the function from `fnMap`, but branches inside the method body (e.g. an `if`) are still reported as uncovered.

Reproduces on `oxc-coverage-instrument@0.3.10`. Same shape as the (closed) #14, but for methods inside object literals.

## Repro

```js
import { instrument } from 'oxc-coverage-instrument'

const code = `const obj = {
  /* v8 ignore next -- @preserve */
  method(x) {
    const y = x.foo
    if (y) {
      y.bar = 1
    }
  },
}`

const m = JSON.parse(instrument(code, 'x.js', { sourceMap: false }).coverageMap)
console.log('functions:', Object.keys(m.fnMap).length)   // 0 (good — pragma respected)
console.log('branches:', Object.keys(m.branchMap).length) // 1 (bad — `if` still tracked)
```

The if-branch ends up at `[0, 0]` and the method's `else` shows as uncovered, even though the whole method was meant to be skipped.

## Expected

When a method is fully ignored, branches and statements inside its body should also be ignored — matching the behaviour for top-level function declarations after the #14 fix.

**Comment by BartWaardenburg** [comment]
Thanks again for the precise report, @simleb. This is fixed in v0.3.11: `ignore next` on object-literal method syntax now suppresses the whole method subtree, including branches inside the method body. I added both Rust integration coverage and N-API smoke coverage for this shape.
